pkg/distro/fedora/distro.go

```diff
@@ -902,17 +902,26 @@ func newDistro(version int) distro.Distro {
 	return &rd
 }
 
-func DistroFactory(idStr string) distro.Distro {
+func DistroIDParser(idStr string) (*distro.ID, error) {
```

Comment:
I was thinking we could rename this to `IDParser()` so it gets called as:
```
id, err := distro.IDParser(idstr)
```
but since it's a function that parses, let's make it a verb phrase:
```
id, err := distro.ParseID(idstr)
```

Reply:
I like your suggestions. I was a bit struggling with the naming for the parser package and functions, so a fresh opinion is very welcomed :slightly_smiling_face:

Reply:
Done.